src/test/java/com/nvs/store/controller/ProductControllerTest.java

```diff
@@ -0,0 +1,132 @@
+package com.nvs.store.controller;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nvs.store.dto.ProductDTO;
+import com.nvs.store.models.product.Product;
+import com.nvs.store.service.ProductService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ProductControllerTest {
```

Comment:
Please discuss and agreed for using the same approach and mocking API.
For ex. to explicitly divide by comment give/when/then blocks or not. 
For mocking to use "doReturn(object).when(mock).method_invocation;" (ex. doReturn(product).when(productService).addProduct(eq(dto));) or to use "when(mock and method execution).thenReturn(object);" (ex. when(productService.getProduct(anyLong())).thenReturn(product);).